core/audits/audit.js

```diff
@@ -339,6 +347,30 @@ class Audit {
     return score;
   }
 
+  /**
+   * @param {LH.Audit.MetricSavings|undefined} metricSavings
+   * @return {LH.Audit.MetricSavings|undefined}
+   */
+  static _normalizeMetricSavings(metricSavings) {
+    if (!metricSavings) return;
+
+    /** @type {LH.Audit.MetricSavings} */
+    const normalizedMetricSavings = {...metricSavings};
+
+    // eslint-disable-next-line max-len
+    for (const key of /** @type {Array<keyof LH.Audit.MetricSavings>} */ (Object.keys(metricSavings))) {
+      const value = metricSavings[key];
+      if (value === undefined) continue;
+
+      const precision = METRIC_SAVINGS_PRECISION[key];
+      if (precision === undefined) continue;
```

Comment:
Oh, because the properties are all optional. You could make `METRIC_SAVINGS_PRECISION` `Record<keyof  LH.Audit.MetricSavings, number>` or `Required<LH.Audit.MetricSavings>` (that would require a precision for INP, but maybe that's good anyways)

Reply:
Good point about INP, I forgot that and will add. With recent type changes we would need to also provide precision for stuff like `SI` and `FMP` which will probably never have metric savings.

Reply:
Kind of a side point, but looking at #15540 now, maybe it's not worth lhr.d.ts knowing about the metric acronyms? There's benefits to the report having to deal with the generic acronym `string` matching (free backwards/forwards compatibility), while probably still enforcing specific acronyms in `core/`.

It seems like the renderer implementation wouldn't have to change, just the types, but overconstraining the types now may require more work down the line. I guess the main weird thing might be `AuditResult.MetricSavings` would then need to be `Record<string, number>` while we'd still want `Audit.ProductBase.metricSavings` to be the enumerated set of acronyms.